src/core/annotation.js

```diff
@@ -661,6 +677,63 @@ class WidgetAnnotation extends Annotation {
     return !!(this.data.fieldFlags & flag);
   }
 
+  /**
+   * Parse widget annotation appearance and extract font information.
+   *
+   * @public
+   * @memberof WidgetAnnotation
+   * @param {number} flag - Hexadecimal representation for an annotation
+   *                        field characteristic
+   * @return {Promise}
+   */
+  parseAppearance(params) {
+    if (!params.pdfManager || !params.pdfManager.pdfDocument) {
+      return Promise.resolve(this);
+    }
+
+    let data = this.data;
+    let self = this;
+
+    let opList = new OperatorList();
+    let appearanceStream = new Stream(stringToBytes(data.defaultAppearance));
+    let formFonts = params.pdfManager.pdfDocument.acroForm.get('DR');
+    opList.acroForm = params.pdfManager.pdfDocument.acroForm;
+    return params.evaluator.getOperatorList({
+      stream: appearanceStream,
+      task: params.task,
+      resources: formFonts,
+      operatorList: opList,
+    }).then(() => {
+      let a = opList.argsArray;
+      let i;
+      for (i = 0; i < opList.fnArray.length; i++) {
+        let fn = opList.fnArray[i];
+        switch (fn | 0) {
+          case OPS.setFont:
+            data.annotationFonts = opList.acroForm.annotationFonts;
+            data.fontRefName = a[i][0];
+            data.fontSize = a[i][1];
+            break;
+          case OPS.setGrayFill:
+            if (a[i].length >= 1) {
+              let gray = Math.round(a[i][0] * 0x100);
+              data.fontColor = Util.makeCssRgb(gray, gray, gray);
```

Comment:
Do this operator and the one below really operate on the font? Their names indicate to me that they work on the rectangle instead, providing gray/RGB fill colors, but I may be wrong here!

Reply:
In this case PDF specs are vague, I didn't find if font color is always specified as RGB or it's possible that gray might be used. I'd rather to keep it just in case.